akka-stream/src/main/scala/akka/stream/impl/ActorPublisher.scala

```diff
@@ -64,8 +64,7 @@ private[akka] class ActorPublisher[T](val impl: ActorRef) extends Publisher[T] {
 
   def takePendingSubscribers(): immutable.Seq[Subscriber[_ >: T]] = {
     val pending = pendingSubscribers.getAndSet(Nil)
-    assert(pending ne null, "takePendingSubscribers must not be called after shutdown")
-    pending.reverse
+    if (pending eq null) Nil else pending.reverse
```

Comment:
I think this is safe. The reason is when `exposedPublisher.shutdown` and `context.stop` is performed. I guess the patch delayed `context.stop`. Anyway, this should be good.